gdpr/full_enforcement.go

```diff
@@ -0,0 +1,89 @@
+package gdpr
+
+import (
+	"github.com/prebid/go-gdpr/api"
+	tcf2 "github.com/prebid/go-gdpr/vendorconsent/tcf2"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	pubRestrictNotAllowed           = 0
+	pubRestrictRequireConsent       = 1
+	pubRestrictRequireLegitInterest = 2
+)
+
+// FullEnforcement determines if legal basis is satisfied for a given purpose and bidder using
+// the TCF2 full enforcement algorithm. The algorithm is a detailed confirmation that reads the
+// GVL, interprets the consent string and performs legal basis analysis necessary to perform a
+// privacy-protected activity.
+// FullEnforcement implements the PurposeEnforcer interface
+type FullEnforcement struct {
+	cfg purposeConfig
+}
+
+// NewFullEnforcement creates a FullEnforcement object
+func NewFullEnforcement(cfg purposeConfig) *FullEnforcement {
+	return &FullEnforcement{
+		cfg: cfg,
+	}
+}
```

Comment:
Why do we need a `NewFullEnforcement` method here? Can we create the `FullEnforcement` directly?

Reply:
We don't need it; we can create it directly.